examples/Makefile.mk

```diff
@@ -0,0 +1,45 @@
+##
+## Copyright (C) by Argonne National Laboratory
+##     See COPYRIGHT in top-level directory
+##
+
+AM_CPPFLAGS += -I$(top_srcdir)/examples
+
+EXTRA_PROGRAMS += $(example_list)
+
+example_list = examples/contig         \
+               examples/vector         \
+               examples/hvector        \
+               examples/indexed_block  \
+               examples/hindexed_block \
+               examples/indexed        \
+               examples/hindexed       \
+               examples/resized        \
+               examples/subarray       \
+               examples/fuf            \
+               examples/iov
+
+example_utils = examples/matrix_util.c
+
+examples_contig_SOURCES         = examples/contig.c $(example_utils)
+examples_contig_LDADD           = libyaksa.la
```

Comment:
I think the LDADD is already handled in the `test/` directory, but that should probably be moved to the top-level `Makefile.am`.  I don't think you need extra `LDADD` rules for each example.

Reply:
If I use the LDADD from the top level Makefile and compile examples before tests it will fail on dtpools dependency

Reply:
I don't think it'll fail.  I think it'll build dtpools too, but yes, I agree that that'd be unnecessary.  Perhaps the specific `LDADD` is better.  I'll retract my comment.